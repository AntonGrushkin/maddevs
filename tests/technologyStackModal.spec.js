import {
  mount
} from '@vue/test-utils';
import technologyStackModal from '@/components/Modals/technology-stack-modal';

describe('Technology stack modal', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = mount(technologyStackModal, {
      stubs: ['ValidationProvider', 'ValidationObserver', 'modal']
    });
  });

  // ------ IMPORTANT ----- //
  test('is a Vue instance', () => {
    expect(wrapper.isVueInstance()).toBeTruthy();
  });
  
  test('renders correctly', () => {
    expect(wrapper.element).toMatchSnapshot();
  });
  // --------------------- //

  test('sets the correct default data', () => {
    expect(typeof technologyStackModal.data).toBe('function');
    const defaultData = technologyStackModal.data();
    expect(
      defaultData.agreeWithPrivacyPolicy &&
      defaultData.agreeToGetMadDevsDiscountOffers
    ).toEqual(false);
    expect(defaultData.inputId).toEqual('technology-stack');
  });

  test('has a functions', () => {
    expect(
      typeof technologyStackModal.methods.getPrivacyCheckboxState && 
      typeof technologyStackModal.methods.getDiscountOffersCheckboxState &&
      typeof technologyStackModal.methods.autosize
    ).toBe('function');
  });

  test('call functions with params and change variables state', () => {
    wrapper.vm.getPrivacyCheckboxState(true);
    wrapper.vm.getDiscountOffersCheckboxState(true);

    expect(
      wrapper.vm.$data.agreeWithPrivacyPolicy &&
      wrapper.vm.$data.agreeToGetMadDevsDiscountOffers
    ).toEqual(true);
  });

  test('autosize function should add value in event key', () => {
    const event = {
      target: {
        style: {
          height: ''
        },
        scrollHeight: 100
      }
    };

    wrapper.vm.autosize(event);
    expect(event.target.style.height).toEqual('100px');
  });
});
