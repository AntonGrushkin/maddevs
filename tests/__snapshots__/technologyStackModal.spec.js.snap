// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Technology stack modal renders correctly 1`] = `
<modal-stub
  height="auto"
  name="technology-stack-modal"
>
  <img
    alt="Close modal"
    class="close-modal"
    src="@/assets/img/common/close-icon.svg"
  />
   
  <perfect-scrollbar-stub
    class="modal_scrollbar"
  >
    <validationobserver-stub />
  </perfect-scrollbar-stub>
</modal-stub>
`;
